src/Akka.Persistence.MongoDb/Snapshot/MongoDbSnapshotStore.cs

```diff
@@ -187,15 +154,38 @@ private SnapshotEntry ToSnapshotEntry(SnapshotMetadata metadata, object snapshot
 
         private SelectedSnapshot ToSelectedSnapshot(SnapshotEntry entry)
         {
-            Type type = null;
 
-            if (!string.IsNullOrEmpty(entry.Manifest))
-                type = Type.GetType(entry.Manifest, throwOnError: true);
 
-            var snapshot = _deserialize(type, entry.Snapshot, entry.Manifest, entry.SerializerId);
+            if (entry.Snapshot is byte[] bytes)
+            {
+                Type type = null;
+
+                if (string.IsNullOrEmpty(entry.Manifest))
+                    type = Type.GetType(entry.Manifest, throwOnError: true);
+
```

Comment:
Also, an *ELSE* seems to be missing to retrieve the SerializerId. Otherwise we could wind up with no `Type`, `SerializerId`, and `Manifest`.